tests/typed_agent.py

```diff
@@ -283,6 +287,17 @@ def my_method(self) -> bool:
 # since deps are not set, they default to `None`, so can't be `int`
 Agent('test', tools=[Tool(foobar_plain)], deps_type=int)  # pyright: ignore[reportArgumentType,reportCallIssue]
 
+# TextOutput with RunContext
+text_output_with_ctx = TextOutput(str_to_regex_with_ctx)
+assert_type(text_output_with_ctx, TextOutput[re.Pattern[str], int])
+Agent('test', output_type=text_output_with_ctx, deps_type=int)
+Agent('test', output_type=text_output_with_ctx, deps_type=bool)  # bool is subclass of int, works with contravariant
+# NOTE: The following don't produce type errors because _OutputSpecItem uses TextOutput[T_co, Any]
```

Comment:
I agree it's worth having a comment like this, but this'll have to be rewritten if my comment above is addressed